cg/models/demultiplex/flow_cell.py

```diff
@@ -77,7 +77,29 @@ def full_name(self) -> str:
     @property
     def sample_sheet_path(self) -> Path:
         """Return sample sheet path."""
-        return Path(self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
+        primary_sample_sheet_path: Path = Path(
+            self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+        )
+
+        unaligned_sample_sheet_path: Path = Path(
+            self.path,
+            DemultiplexingDirsAndFiles.UNALIGNED_DIR_NAME,
+            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+        )
+
+        root_sample_sheet_path: Path = Path(
+            self.path,
+            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+        )
+
+        if primary_sample_sheet_path.exists():
+            return primary_sample_sheet_path
+        elif unaligned_sample_sheet_path.exists():
+            return unaligned_sample_sheet_path
+        elif root_sample_sheet_path.exists():
+            return root_sample_sheet_path
+
+        return primary_sample_sheet_path
```

Comment:
```suggestion

sample_sheet_paths: Optional[Path] = [
        primary_sample_sheet_path: Path = Path(
            self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
        ),

        unaligned_sample_sheet_path: Path = Path(
            self.path,
            DemultiplexingDirsAndFiles.UNALIGNED_DIR_NAME,
            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
        ),

        root_sample_sheet_path: Path = Path(
            self.path,
            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
        )]
        
        for sample_sheet_path in sample_sheet_paths:
            if sample_sheet_path.exsists():
                return sample_sheet_path
      # Some error handling/ warning maybe?
```

Maybe something like this? I don't think it matters which sample sheet you get since they are all the same. So as soon as one exists you can return it. right?

Reply:
I like that, a bit more readable maybe